File: BuildServer/game/model/ArmsInfo.py
# coding=utf-8
class ArmsInfo:

    def __init__(self):
        self.moveStep = 1
        self.ifStraight = False
        self.attackRange = 1
        self.maxTl = 1


def get_arms_info_by_name(arms, yl):
    result = ArmsInfo()
    if arms == u"步":
        result.moveStep = 2
        result.ifStraight = False
        result.attackRange = 1
        result.maxTl = 25 + yl * 2
    elif arms == u"骑":
        result.moveStep = 3
        result.ifStraight = True
        result.attackRange = 1
        result.maxTl = 20 + yl * 2
    elif arms == u"弓":
        result.moveStep = 2
        result.ifStraight = False
        result.attackRange = 2
        result.maxTl = 17 + yl * 2
    elif arms == u"器":
        result.moveStep = 2
        result.ifStraight = True
        result.attackRange = 1
        result.maxTl = 25 + yl * 2

    return result

File: EgretSource/src/model/general/ArmsInfo.ts

class ArmsInfo {
  moveStep: number;
  ifStraight: boolean;
  attackRange: number;
  maxTl: number;

  static getArmsInfoByName = function (arms: string, yl: number): ArmsInfo {
    let result = new ArmsInfo();
    switch (arms) {
      case "步":
        result = {
          moveStep: 2,
          ifStraight: false,
          attackRange: 1,
          maxTl: 25 + yl * 2,
        };
        break;
      case "骑":
        result = {
          moveStep: 3,
          ifStraight: false,
          attackRange: 1,
          maxTl: 20 + yl * 2,
        };
        break;
      case "弓":
        result = {
          moveStep: 2,
          ifStraight: false,
          attackRange: 2,
          maxTl: 15 + yl * 2,
        };
        break;
      case "器":
        result = {
          moveStep: 2,
          ifStraight: true,
          attackRange: 1,
          maxTl: 25 + yl * 2,
        };
        break;
    }
    return result;
  };
}

